core/src/main/java/io/questdb/cairo/TableWriter.java

```diff
@@ -1460,39 +1460,41 @@ public boolean removePartition(long timestamp) {
             // removing active partition
 
             if (index == 0) {
-                // removing the very last partition is equivalent to truncating the table
-                truncate();
-                return true;
-            }
+                // the last partition
+                columnVersionWriter.removePartition(timestamp);
+                txWriter.truncate(columnVersionWriter.getVersion());
+                closeActivePartition(true);
+                row = regularRow;
+                rowAction = ROW_ACTION_OPEN_PARTITION;
```

Comment:
Should we call safeDeletePartitionDir(timestamp, partitionNameTxn); here?

Reply:
yes, I will extract the call so that it is called regardless of the case.